### PR TITLE
[WFCORE-2025] Correct the priority when selecting the SSLContext to use with the CLI.

### PR DESCRIPTION
**Pull request by darranl**
:bangbang: Commits from this branch are used in numerous other topic branches so please do not cherry-pick or rebase. :bangbang:

The SSLContext priority order becomes: -
 1. The SSLContext from jboss-cli.xml
 2. An SSLContext from wildfly-config.xml
 3. The default SSLContext that can prompt for trust decisions.

The following issue can be resolved with this PR: -
  https://issues.jboss.org/browse/WFCORE-2025

### CODE REVIEW

File: cli/src/main/java/org/jboss/as/cli/impl/CLIModelControllerClient.java [hunk 1]
```diff
@@ -146,6 +150,23 @@ public void close() throws IOException {
             throw new IOException("Failed to create URI" , e);
         }
 
+        if (sslContext != null && fallbackSslContext == false) {
```
Comment (jfdenise) [inline cli/src/main/java/org/jboss/as/cli/impl/CLIModelControllerClient.java]: perhaps use !fallbackSslContext

File: cli/src/main/java/org/jboss/as/cli/impl/CommandContextImpl.java [hunk 2]
```diff
@@ -252,6 +252,8 @@
     private final char[] password;
     /** flag to disable the local authentication mechanism */
     private final boolean disableLocalAuth;
+    /** Flag to indicate that the SSLContext was the defailt context and not configured */
```
Comment (jfdenise) [inline cli/src/main/java/org/jboss/as/cli/impl/CommandContextImpl.java]: defailt ==> default

File: cli/src/main/java/org/jboss/as/cli/impl/CLIModelControllerClient.java (hunk 1 above)
Comment (darranl) [inline cli/src/main/java/org/jboss/as/cli/impl/CLIModelControllerClient.java]: Never been a fan of '!' especially after wasted time on support cases.

File: cli/src/main/java/org/jboss/as/cli/impl/CommandContextImpl.java (hunk 2 above)
Comment (darranl) [inline cli/src/main/java/org/jboss/as/cli/impl/CommandContextImpl.java]: Lets leave that and pick it up next time we are here.

File: protocol/src/main/java/org/jboss/as/protocol/ProtocolConnectionConfiguration.java [hunk 3]
```diff
@@ -103,12 +102,13 @@ public void setEndpoint(Endpoint endpoint) {
...
+    @Deprecated
     public OptionMap getOptionMap() {
-        return optionMap;
+        return OptionMap.EMPTY;
     }
 
+    @Deprecated
     public void setOptionMap(OptionMap optionMap) {
-        this.optionMap = optionMap;
     }
```
Comment (bstansberry) [inline protocol/src/main/java/org/jboss/as/protocol/ProtocolConnectionConfiguration.java]: @darranl this change to how optionMap is handled looks to me to be a breaking change. RemoteDomainConnectionService.start is using this to configure heartbeats and a read timeout.

**Pull request merged**